static/js/stream_create.js

```diff
@@ -134,6 +134,44 @@ function update_announce_stream_state() {
     $('#announce-new-stream').show();
 }
 
+function get_principals() {
+    return _.map(
+        $("#stream_creation_form input:checkbox[name=user]:checked"),
+        function (elem) {
+            return $(elem).val();
+        }
+    );
+}
+
+function create_stream() {
+    var stream = $.trim($("#create_stream_name").val());
+    var description = $.trim($("#create_stream_description").val());
+    var is_invite_only = $('#stream_creation_form input[name=privacy]:checked').val() === "invite-only";
+    var principals = get_principals();
+
+    // You are always subscribed to streams you create.
+    principals.push(people.my_current_email());
+
+    created_stream = stream;
+
+    var announce = (!!page_params.notifications_stream &&
+        $('#announce-new-stream input').prop('checked'));
+
+    ajaxSubscribeForCreation(stream,
+        description,
+        principals,
+        is_invite_only,
+        announce
+    );
+}
```

Comment:
This would probably be more readable if you did it in 2 commits: One to refactor to create this function, and a second to add the new feature.